src/libpriv/rpmostree-core.cxx

```diff
@@ -1916,9 +1916,43 @@ find_locked_packages (RpmOstreeContext *self,
 
   g_autoptr(GPtrArray) pkgs =
     g_ptr_array_new_with_free_func ((GDestroyNotify)g_object_unref);
+
+  auto locked_src_pkgs = (*self->lockfile)->get_locked_src_packages();
+  g_autoptr(GHashTable) locked_src_pkgnames =
```

Comment:
I think this could be `std::set<rust::String>` too.  But fine as is.

Reply:
Yup nice, avoids a bunch of allocation. (Still need to also circle back on all this to have the shared `LockedPackage` struct just use references instead of copying.)